python/pyspark/serializers.py

```diff
@@ -206,13 +206,19 @@ def load_stream(self, stream):
         for batch in self.serializer.load_stream(stream):
             yield batch
 
-        # load the batch order indices
-        num = read_int(stream)
-        batch_order = []
-        for i in xrange(num):
-            index = read_int(stream)
-            batch_order.append(index)
-        yield batch_order
+        # check success
+        success = read_bool(stream)
+        if success:
+            # load the batch order indices
+            num = read_int(stream)
+            batch_order = []
+            for i in xrange(num):
+                index = read_int(stream)
+                batch_order.append(index)
+            yield batch_order
+        else:
+            error_msg = UTF8Deserializer().loads(stream)
+            raise RuntimeError("An error occurred while collecting: {}".format(error_msg))
```

Comment:
I think the error message would be a little better like this "An error occurred while calling ArrowCollectSerializer.load_stream: {}".format(error_msg)